### ISSUE TITLE
Undefined $SED in configure

### ISSUE THREAD

**Issue by albandil** [open-mpi/ompi#10392]
When I configure the `v5.0.0rc7` tag from the repository on openSUSE Tumbleweed (with up to date submodules and *autogen.pl* executed), I see this output:
```
(...)
*** Fortran compiler
checking whether the compiler supports GNU Fortran... yes
checking whether gfortran-12 accepts -g... yes
checking for BSD- or MS-compatible name lister (nm)... ../../configure: line 32694: 1q: command not found
../../configure: line 32700: 1q: command not found
../../configure: line 32694: 1q: command not found
../../configure: line 32700: 1q: command not found
../../configure: line 32694: 1q: command not found
../../configure: line 32700: 1q: command not found
/usr/bin/nm
checking the name lister (/usr/bin/nm) interface... BSD nm
(...)
```
The lines 32694 and 32700 are similar and both contain `$SED '1q'`. The content of the variable `$SED` is apparently empty. Only when I do
```
SED=sed ./configure
```
then the configuration finishes without complaints. Open MPI, PMIX, as well as PRTE seem to be affected by the same problem; I see the same set of command-not-found messages three times.

**Comment by ggouaillardet** [comment]
Thanks for the report!

some `AC_PROG_SED` calls are missing in various `configure.ac`

I also noted lex library detection is busted on Tumbleweed:
`/usr/lib64/libfl.so` expects a `yylex` symbol but it is not defined in the `conftest.c` generated by autotools.

I will issue the PRs and check the flex detection issue later on.

**Comment by awlauria** [comment]
Fixed by #10400 (main) and #10406 (v5.0.x)
